Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx

```diff
@@ -1484,7 +1484,8 @@ bool JPEG2000Codec::GetHeaderInfo(const char * dummy_buffer, size_t buf_size, Tr
     &image) ? true : false;
   if(!bResult)
   {
-  opj_stream_destroy(cio);
+    opj_destroy_codec(dinfo);
+    opj_stream_destroy(cio);
     return false;
```

Comment:
Should these structures be "destroyed" with each and every return or exit (possibly by an exception) out of the function? Then maybe the C++ idiom [RAII](https://en.cppreference.com/w/cpp/language/raii) would be preferable.  For example, `dinfo` and `cio` could be declared as `unique_ptr` with a custom deleter function, being `opj_destroy_codec` or `opj_stream_destroy`, respectively.

Well, I said so already at https://github.com/InsightSoftwareConsortium/ITK/pull/5308#discussion_r2060086431

Reply:
This is the same fix as #5308, just targeting release branch. I guess that review comments should be made in the other PR.